packages/node_modules/@webex/webex-core/src/lib/services/service-catalog.js

```diff
@@ -106,6 +108,110 @@ const ServiceCatalog = AmpState.extend({
     return this;
   },
 
+  /**
+   * Search over all service groups to find a cluster id based
+   * on a given url.
+   * @param {string} url - Must be parsable by `Url`
+   * @returns {string} - ClusterId of a given url
+   */
+  findClusterId(url) {
+    const incomingUrlObj = Url.parse(url);
+    let clusterId;
+    let serviceUrlObj;
+
+    Object.keys(this.serviceGroups).forEach((key) => {
+      this.serviceGroups[key].forEach((service) => {
+        serviceUrlObj = Url.parse(service.defaultUrl);
+
+        if (serviceUrlObj.host === incomingUrlObj.host &&
+          service.hosts.length > 0) {
+          clusterId = service.hosts[0].id;
+        }
+
+        service.hosts.forEach((host) => {
+          if (incomingUrlObj.host === host.host && host.id) {
+            clusterId = host.id;
+          }
+        });
+      });
+    });
+
+    return clusterId;
+  },
+
+  /**
+   * Search over all service groups and return a service value from a provided
+   * clusterId. Currently, this method will return either a service name, or a
+   * service url depending on the `value` parameter. If the `value` parameter
+   * is set to `name`, it will return a service name to be utilized within the
+   * Services plugin methods.
+   * @param {object} params
+   * @param {string} params.clusterId - clusterId of found service
+   * @param {boolean} [params.priorityHost] - returns priority host url if true
+   * @param {string} [params.serviceGroup] - specify service group
+   * @param {string} [params.value] - select value of service to return
```

Comment:
Does it make things easier if we returned an object with both the name and url instead of having to select one or the other via param?

Reply:
addressed.